src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -1054,17 +1065,22 @@ public void RunIdleOperations()
 
             foreach (var indexName in indexesToRemove)
             {
-                await TryDeleteIndexIfExists(indexName);
-                if (_logger.IsInfoEnabled)
-                    _logger.Info($"Deleted index '{indexName}' because it is surpassed.");
+                try
+                {
+                    await TryDeleteIndexIfExists(indexName);
+                    if (_logger.IsInfoEnabled)
+                        _logger.Info($"Deleted index '{indexName}' because it is surpassed.");
+                }
+                catch (Exception e)
+                {
+                    if (_logger.IsOperationsEnabled)
+                        _logger.Operations($"Could not delete surpassed index '{indexName}'.", e);
+                }
             }
         }
 
         private async Task HandleUnusedAutoIndexes()
```

Comment:
I'm not sure how correct this is going to be.  What happens if queries are made on a _different_ node, and not this one?

Reply:
Might actually be better to move this entirely to the cluster observer?

Reply:
Created a separate issue for that: http://issues.hibernatingrhinos.com/issue/RavenDB-11206